spec/models/change_service_event_spec.rb

```diff
@@ -21,6 +21,19 @@
       expect(change_service_event.data['name']).to eql(new_service_name)
     end
   end
+  context '#trigger_publish_event' do
+    it 'when change service event occurs it triggers a publish event' do
+      component_1 = create(:sp_component)
+      event = create(:assign_sp_component_to_service_event, service: service, sp_component_id: component_1.id)
+      updated_entity_id = 'https://changed-entity-id1'
+      service.assign_attributes(entity_id: updated_entity_id)
+      change_service_event = create(:change_service_event, service: service)
+
+      resulting_event = PublishServicesMetadataEvent.order(:created_at).last
```

Comment:
I found using order or last or any variant of this, problematic in the past they work at first but could start to fail later; this might well be just fine but I think this 
```
resulting_event = PublishServicesMetadataEvent.where("data->>'event_id' = ?", change_service_event.id).first
```
Is more future proof. Just a recommendation.